packages/components/src/components/assets/CoinLogo/CoinLogo.tsx

```diff
@@ -1,38 +1,130 @@
+    const fileName = contractAddress ? `${coingeckoId}_${contractAddress}` : coingeckoId;
+    const assetUrl = fileName ? `${UPDATED_ICONS_LIST_URL_BASE}${fileName}@2x.webp` : '';
+    const isAssetUrlAccessible = useCheckUrlAccessibility(assetUrl);
+    const iconUrl = fileName && isAssetUrlAccessible ? assetUrl : undefined;
+
+    return iconUrl;
+};
+
 export interface CoinLogoProps extends ImgHTMLAttributes<HTMLImageElement> {
-    symbol: CoinType;
+    symbol?: CoinType;
     className?: string;
     size?: number;
     index?: number;
+    coingeckoId?: string;
+    contractAddress?: string;
+    quality?: Quality;
 }
 
-const SvgWrapper = styled.div<Omit<CoinLogoProps, 'symbol'>>`
+const Wrapper = styled.div<{ $size: number }>`
     display: inline-block;
-    width: ${({ size }) => size}px;
-    height: ${({ size }) => size}px;
+    width: ${({ $size }) => $size}px;
+    height: ${({ $size }) => $size}px;
+    margin: ${spacingsPx.xxs};
 
     div {
-        width: ${({ size }) => size}px;
-        height: ${({ size }) => size}px;
-        line-height: ${({ size }) => size}px;
+        width: ${({ $size }) => $size}px;
+        height: ${({ $size }) => $size}px;
+        line-height: ${({ $size }) => $size}px;
     }
 `;
 
-export const CoinLogo = ({ symbol, className, size = 32, ...rest }: CoinLogoProps) => (
-    <SvgWrapper className={className} size={size} {...rest}>
-        <ReactSVG
-            src={COINS[symbol]}
-            beforeInjection={svg => {
-                svg.setAttribute('width', `${size}px`);
-                svg.setAttribute('height', `${size}px`);
-            }}
-            loading={() => <span className="loading" />}
-        />
-    </SvgWrapper>
+const StyledIconImage = styled.img<{ $size: number }>(
+    ({ $size }) => `
+        width: ${$size}px;
+        height: ${$size}px;
+    `,
+);
+
+type SVGCoinLogoProps = {
+    $symbol: CoinType;
+    $size?: number;
+};
+
+const SVGCoinLogo = ({ $symbol, $size = 32 }: SVGCoinLogoProps) => (
+    <ReactSVG
+        src={$symbol}
+        beforeInjection={svg => {
+            svg.setAttribute('width', `${$size}px`);
+            svg.setAttribute('height', `${$size}px`);
+        }}
+    />
 );
+
+export const CoinLogo = ({
+    symbol,
+    className,
+    size = 32,
+    coingeckoId = '',
+    contractAddress,
+    quality = 'medium',
+    ...rest
+}: CoinLogoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+
+    const iconUrl = useAssetUrl(coingeckoId, contractAddress);
+    const firstCharacter = coingeckoId && coingeckoId[0].toUpperCase();
+    const coinLogoSize = !symbol ? QUALITY_SIZE[quality] : size;
+
+    const logo = iconUrl ? (
+        <>
+            {isLoading && <span className="loading">Loading...</span>}
+            <StyledIconImage
+                src={iconUrl}
+                $size={coinLogoSize}
+                onLoad={() => setIsLoading(false)}
+                onError={() => setIsLoading(false)}
```

Comment:
What about unify this function? 

Maybe something like this?
```
onLoad={hideLoading}
onError={hideLoading}
```

Reply:
Awesome! How couldn't I have seen that earlier? :D